main.py

```diff
@@ -0,0 +1,166 @@
+# The gui is based in a tkinter tutorial found at: https://www.geeksforgeeks.org/python-tkinter-entry-widget/
+import tkinter as tk
+from tkinter import messagebox
+from math import pow, cos, pi
+from aproximation_rules import mid_rule, trap_rule, simpsons_rule
+
+my_func = ''
+interval = [] # x values between which we want to calculate the area (usually refered as 'a' and 'b')
+n = 0 #  number of rectangles/partitions
+
+def f_of_x(x_values):
+    # TODO: take a function from the user
+    result = []
+    for i in range(0, len(x_values)):
+        x = x_values[i]
+        result.append( pow(1+(cos(x)), 1/3) )
+    
+    return result
+
+# GUI------------------------------------------------------------------
+def handle_focus_in(_):
+    interval_entry.delete(0, tk.END)
+    interval_entry.config(fg='black')
+
+def handle_focus_out(_):
+    interval_entry.delete(0, tk.END)
+    interval_entry.config(fg='grey')
+    interval_entry.insert(0, "Simply enter a,b")
+
+def handle_enter(txt):
+    print(interval_entry.get())
+    handle_focus_out('')
+
+def clear_entries():
+    
+    func_var.set("") 
+    interval_var.set("")
+    n_var.set("") 
+
+    interval_entry.delete(0, 'end')
+    interval_entry.insert(0, 'Simply enter a,b')
+
+def prompt_error():
+    tk.messagebox.showerror(title='Input error', message='Invalid input. Please try again')
+    clear_entries()
+
+def compute(): 
+   
+    try:
+        my_func = func_entry.get()
+        print('func ok') 
+
+        for word in (interval_entry.get()).split(','):
+            if word.isdigit():
+                interval.append(int(word))
+        print('inetrval ok') 
+        n = n_entry.get()
+        n = int(n)
+
+        if len(interval) != 2:
+            prompt_error()
+    except:

```

Comment:
maybe separate the GUI functions into their own file so it's easier to read and follow